module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -897,12 +899,23 @@ public function getMyTransactions($patron)
             '/circulation/loans',
             $query
         ) as $trans) {
-            $date = date_create($trans->dueDate);
+            $date = new DateTime($trans->dueDate, new DateTimeZone('UTC'));
+            $loc = (new DateTime)->getTimezone();
+            $date->setTimezone($loc);
+
+            $dueStatus = false;
+            $tmpDueDate = strtotime($trans->dueDate);
```

Comment:
Should you derive $tmpDueDate from $date above, instead of going back to the raw $trans->dueDate? Just wondering if this impacts the time zone handling.

Reply:
You are probably correct that it will work either way, but I suspect switching it to use `$date->getTimestamp()` would be more efficient since it would eliminate the need to parse the date twice, which I'm sure is a more expensive operation than fetching an already-calculated timestamp.